Modules/_cursesmodule.c

```diff
@@ -3967,49 +4010,32 @@ _curses_qiflush_impl(PyObject *module, int flag)
 /* Internal helper used for updating curses.LINES, curses.COLS, _curses.LINES
  * and _curses.COLS */
 #if defined(HAVE_CURSES_RESIZETERM) || defined(HAVE_CURSES_RESIZE_TERM)
-static int
-update_lines_cols(void)
+static int /* 1 on success, 0 on failure */
+update_lines_cols(PyObject *private_module)
 {
-    PyObject *o;
-    PyObject *m = PyImport_ImportModule("curses");
-
-    if (!m)
-        return 0;
-
-    o = PyLong_FromLong(LINES);
-    if (!o) {
-        Py_DECREF(m);
-        return 0;
-    }
-    if (PyObject_SetAttrString(m, "LINES", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
+    PyObject *exposed_module = PyImport_ImportModule("curses");
+    if (exposed_module == NULL) {
         return 0;
     }
-    if (PyDict_SetItemString(ModDict, "LINES", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
-        return 0;
-    }
-    Py_DECREF(o);
-    o = PyLong_FromLong(COLS);
-    if (!o) {
-        Py_DECREF(m);
-        return 0;
-    }
-    if (PyObject_SetAttrString(m, "COLS", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
-        return 0;
-    }
-    if (PyDict_SetItemString(ModDict, "COLS", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
-        return 0;
-    }
-    Py_DECREF(o);
-    Py_DECREF(m);
+#define MODULE_ADD_INT_CONSTANT(MODULE, NAME, VALUE)                        \
+    do {                                                                    \
+        if (PyModule_AddIntConstant((MODULE), (NAME), (long)(VALUE)) < 0) { \
```

Comment:
I think the compiler won't complain but I'm not entirely sure since we use macros. If an implementation uses typed global variables, we might want to force the cast (I've seen both usages of explicit casting and sometimes not but I don't think the cast hurts)